applications/ExaquteSandboxApplication/tests/cpp_tests/test_metric_process.cpp

```diff
@@ -0,0 +1,466 @@
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//                   Riccardo Tosi
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "includes/kratos_flags.h"
+// #include "includes/gid_io.h"
+#include "containers/model.h"
+
+/* Processes */
+#include "processes/compute_nodal_gradient_process.h"
+#include "custom_processes/metrics_divergencefree_process.h"
+#include "meshing_application_variables.h"
+
+namespace Kratos
+{
+    namespace Testing
+    {
+        typedef Node<3> NodeType;
+
+//         void GiDIODebugMetric(ModelPart& rModelPart)
+//         {
+//             GidIO<> gid_io("TEST_METRIC", GiD_PostBinary, SingleFile, WriteUndeformed,  WriteElementsOnly);
+//             const int nl_iter = rModelPart.GetProcessInfo()[NL_ITERATION_NUMBER];
+//             const double label = static_cast<double>(nl_iter);
+//
+//             gid_io.InitializeMesh(label);
+//             gid_io.WriteMesh(rModelPart.GetMesh());
+//             gid_io.FinalizeMesh();
+//             gid_io.InitializeResults(label, rModelPart.GetMesh());
+//             gid_io.WriteNodalResults(DISTANCE, rModelPart.Nodes(), label, 0);
+//             gid_io.WriteNodalResults(DISTANCE_GRADIENT, rModelPart.Nodes(), label, 0);
+//             gid_io.WriteNodalResultsNonHistorical(METRIC_TENSOR_2D, rModelPart.Nodes(), label);
+// //             gid_io.WriteNodalResultsNonHistorical(METRIC_TENSOR_3D, rModelPart.Nodes(), label); // NOTE: 6 components not suported, update
+//         }
+//
+//         void GiDIODebugMetricSPR(ModelPart& rModelPart)
+//         {
+//             GidIO<> gid_io("TEST_METRIC_SPR", GiD_PostBinary, SingleFile, WriteUndeformed,  WriteElementsOnly);
+//             const int nl_iter = rModelPart.GetProcessInfo()[NL_ITERATION_NUMBER];
+//             const double label = static_cast<double>(nl_iter);
+//
+//             gid_io.InitializeMesh(label);
+//             gid_io.WriteMesh(rModelPart.GetMesh());
+//             gid_io.FinalizeMesh();
+//             gid_io.InitializeResults(label, rModelPart.GetMesh());
+//             gid_io.WriteNodalResults(DISPLACEMENT, rModelPart.Nodes(), label, 0);
+//             gid_io.PrintOnGaussPoints(ERROR_INTEGRATION_POINT, rModelPart, label);
+//             gid_io.PrintOnGaussPoints(CAUCHY_STRESS_VECTOR, rModelPart, label);
+//             gid_io.PrintOnGaussPoints(STRAIN_ENERGY, rModelPart, label);
+//             gid_io.WriteNodalResultsNonHistorical(METRIC_TENSOR_2D, rModelPart.Nodes(), label);
+// //             gid_io.WriteNodalResultsNonHistorical(METRIC_TENSOR_3D, rModelPart.Nodes(), label); // NOTE: 6 components not suported, update
+//         }
+
+        void Create2DGeometry(ModelPart& rModelPart, const std::string& ElementName)
```

Comment:
We should move all this geometries to an auxiliar external file, otherwise we are always copy pasting... for a future PR